pkg/xds/server/components.go

```diff
@@ -117,6 +132,27 @@ func DefaultDataplaneSyncTracker(rt core_runtime.Runtime, reconciler SnapshotRec
 					return err
 				}
 
+				if dataplane.Spec.GetNetworking().GetIngress() != nil {
+					dataplanes, err := ingress.GetAllDataplanes(rt.ReadOnlyResourceManager())
+					if err != nil {
+						return err
+					}
+					// update Ingress
+					dataplane.Spec.Networking.Ingress = ingress.GetIngressByDataplanes(dataplanes)
+					if err := rt.ResourceManager().Update(ctx, dataplane); err != nil {
```

Comment:
Can we do some kind of diff here so we check if we actually change anything? I'd want to avoid updating resource in DB every 1s if nothing changed